test/mpi/dtpools/src/dtpools.c

```diff
@@ -90,23 +90,21 @@ int DTP_pool_free(DTP_pool_s dtp)
  *
  * dtp:         datatype pool handle
  * obj:         created object handle
+ * maxbufsize:  maximum buffer size allowed
  */
 int DTP_obj_create(DTP_pool_s dtp, DTP_obj_s * obj, MPI_Aint maxbufsize)
 {
     int rc = DTP_SUCCESS;
     DTPI_pool_s *dtpi = dtp.priv;
     int max_tree_depth;
+    int max_attempts = 100; /* set the max number of object create
+                               attempts to 100 and fail if exceed */
```

Comment:
Can we modify this with an environment variable?

Reply:
sure